Source/WebCore/loader/cache/CachedResourceLoader.cpp

```diff
@@ -1052,30 +1062,30 @@ ResourceErrorOr<CachedResourceHandle<CachedResource>> CachedResourceLoader::requ
             return makeUnexpected(ResourceError { errorDomainWebKitInternal, 0, url, "Resource blocked by content blocker"_s, ResourceError::Type::AccessControl });
         }
 #endif
-        if (!madeHTTPS && type == CachedResource::Type::MainResource && frame.loader().shouldUpgradeRequestforHTTPSOnly(frame.document() ? frame.document()->url() : URL { }, request.resourceRequest()) && m_documentLoader->frameLoader())
-            return makeUnexpected(m_documentLoader->frameLoader()->client().httpNavigationWithHTTPSOnlyError(request.resourceRequest()));
+        if (!madeHTTPS && type == CachedResource::Type::MainResource && frame->checkedLoader()->shouldUpgradeRequestforHTTPSOnly(frame->document() ? frame->document()->url() : URL { }, request.resourceRequest()) && m_documentLoader->frameLoader())
+            return makeUnexpected(protectedDocumentLoader()->checkedFrameLoader()->client().httpNavigationWithHTTPSOnlyError(request.resourceRequest()));
 
         if (madeHTTPS
             && type == CachedResource::Type::MainResource
             && m_documentLoader->isLoadingMainResource()) {
             // This is to make sure the correct 'new' URL shows in the location bar.
-            m_documentLoader->frameLoader()->client().dispatchDidChangeProvisionalURL();
+            protectedDocumentLoader()->checkedFrameLoader()->client().dispatchDidChangeProvisionalURL();
         }
         url = request.resourceRequest().url(); // The content extension could have changed it from http to https.
         url = MemoryCache::removeFragmentIdentifierIfNeeded(url); // Might need to remove fragment identifier again.
 
         if (!m_documentLoader->customHeaderFields().isEmpty()) {
             bool sameOriginRequest = false;
-            auto requestedOrigin = SecurityOrigin::create(url);
+            Ref requestedOrigin = SecurityOrigin::create(url);
             if (type == CachedResource::Type::MainResource) {
-                auto* localMainFrame = dynamicDowncast<LocalFrame>(frame.mainFrame());
-                if (frame.isMainFrame())
+                RefPtr localMainFrame = dynamicDowncast<LocalFrame>(frame->mainFrame());
+                if (frame->isMainFrame())
```

Comment:
The way we use `mainFrame` here (as both the true main frame, and the local "main frame" of an iframe is so confusing. (I know this isn't your new code -- it just is jarring to read).

Reply:
Perhaps `mainFrame` should be called `localFrame` or something.

Reply:
This is because of site isolation. The main frame may be local or remote (in another process) once you have out-of-process iframes. It does add some complexity.